### ISSUE TITLE
More horizontal spacing per layer of child nodes in scene tree

### ISSUE THREAD

**Issue by GuyUnger** [godotengine/godot-proposals#3317]
### Describe the project you are working on

NA

### Describe the problem or limitation you are having in your project

It's hard to see the child parent relations in the scene tree

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Just add more spacing. There are not many things where I prefer unity's interface, but this is definitely one of them.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

mockup (the line connections are removed because it was quicker to mockup, but they can stay IMO)
![more tree space](https://user-images.githubusercontent.com/7023847/133918146-4a39c0eb-470d-41d3-be1d-5b659c9a3fa3.png)

example of how it looks in unity
![brave_utlmyEPuzY](https://user-images.githubusercontent.com/7023847/133918155-5647866c-f054-4d6e-86b0-4f9b13be54cf.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

NA

### Is there a reason why this should be core and not an add-on in the asset library?

NA

**Comment by zinnschlag** [comment]
Adding more spacing will also increase the amount of screen space required. I am not opposed to the proposal. But simply increasing the spacing won't do, since the optimal value will depend on factors like the screen size of the individual user and the complexity of the tree. If this is changed it should be turned into an editor user setting.

**Comment by GuyUnger** [comment]
Yeah that's what I thought too (should've mentioned it in the proposal), but I think the default value can be increased too. I don't think anyone complains about screen space in unity. Godot has 9 pixels of spacing where unity has 13 and it's just better IMO. And an extra option for people with the screen space would be amazing

**Comment by pycbouh** [comment]
> I don't think anyone complains about screen space in unity.

That's, frankly, irrelevant. We care about people with laptops and smaller screens and prefer that the UI works for them out of the box. It's much harder to adjust the UI to be smaller when it doesn't fit your screen, than to adjust it to be more spacious when it does.

**Comment by Calinou** [comment]
We could adjust the indentation level dynamically depending on the scene tree dock's width, but this may still not be enough for deeply nested trees or trees with long node names.

**Comment by pycbouh** [comment]
> I would prefer something that works automatically rather than adding yet another editor setting.

We could reuse the additional spacing setting for the editor trees, maybe?

**Comment by Calinou** [comment]
> We could reuse the additional spacing setting for the editor trees, maybe?

That sounds good to me, but I don't think OP wants buttons to use extra spacing.

**Comment by GuyUnger** [comment]
> That's, frankly, irrelevant. 

if the trade-off is between readability and screen space, looking at software with many users that goes more in the direction of readability and in practice doesn't seem to have problems with space, that's not irrelevant

**Comment by junkajii** [comment]
> ...But simply increasing the spacing won't do, since the optimal value will depend on factors like the screen size of the individual user and the complexity of the tree...

For me, I can see the ideal alignment value, just align the next icon to the previous Node text.

![Untitled-1-gd](https://user-images.githubusercontent.com/25551525/133943465-cc9e0c2a-5479-4297-8143-22b5e78b9d5d.png)

**Comment by GuyUnger** [comment]
@junkajii for me that would be enough spacing! having it aligned like that looks much more clean and orderly too

**Comment by LightningAA** [comment]
> Describe the problem or limitation you are having in your project
> It's hard to see the child parent relations in the scene tree

It only affects selected nodes, but https://github.com/godotengine/godot/pull/48546 helps with that.

**Comment by jcostello** [comment]
> > ...But simply increasing the spacing won't do, since the optimal value will depend on factors like the screen size of the individual user and the complexity of the tree...
> 
> For me, I can see the ideal alignment value, just align the next icon to the previous Node text.
> 
> ![Untitled-1-gd](https://user-images.githubusercontent.com/25551525/133943465-cc9e0c2a-5479-4297-8143-22b5e78b9d5d.png)

There is a problem there in the example. The spacing suggest that AnimationPlayer and AnimationTree are nested and child of CollitionShape2D where in fact they are child of Player at the same level

**Comment by EricEzaM** [comment]
Am I missing something here? Make a custom theme for your editor `Tree / Constants / Item Margin` and set it to about 22 to 24. Unless you mean you _only_ want this for the SceneTreeDock, in which case yes, it would need to be done in the source code.

**Comment by Zireael07** [comment]
99% sure that the askers want it ONLY for scene tree dock - the defaults are fine for other tree views imho.

**Comment by JoanPotatoes2021** [comment]
Could we expose a customizable property to the editor settings?

**Comment by pycbouh** [comment]
Technically, yes, we absolutely can, but we also don't want to overburden users with niche settings.
